lib/ansible/modules/net_tools/ip_link_device_attribute.py

```diff
@@ -0,0 +1,536 @@
+        type: str
+      group:
+        description: Group of the interface.
+        type: str
+      arp:
+        description: State of ARP protocol for the interface.
+        type: bool
+      mtu:
+        description: MTU value for the interface.
+        type: int
+      multicast:
+        description: State of mutlicast for the interface.
+        type: bool
+      promisc:
+        description: State of promisc mode for the interface.
+        type: bool
+      txqueuelen:
+        description: Length of the TX queue.
+        type: int
+      state:
+        description: Administrative state for interface, C(up) or C(down).
+        type: str
+      address:
+        description: L2 address fof the interface
+                     (may be C(None), if the interface can not have one).
+        type: str
+      broadcast:
+        description: L2 broadcast address for the interface
+                     (may be C(None), if the interface can not have one).
+        type: str
+      alias:
+        description: Alias for the interface (C(None) if not present).
+        type: str
+"""
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_text
+
+
+__metaclass__ = type
+
+
+class Link(object):
+    """Interface to 'link' object."""
+
+    params_list = [  # module paramtes needed special treatment
+        'group_id', 'namespace', 'name', 'netns'
+    ]
+    knob_cmds = {  # attributes we may change by calling 'ip'
+        'address': lambda addr: ['address', str(addr)],
+        'alias': lambda alias: ['alias', str(alias)],
+        'arp': lambda is_arp: ['arp', ['off', 'on'][is_arp]],
+        'broadcast': lambda addr: ['broadcast', addr],
+        'group': lambda group: ['group', str(group)],
+        'mtu': lambda mtu: ['mtu', str(mtu)],
+        'multicast': lambda is_mc: ['multicast', ['off', 'on'][is_mc]],
+        'promisc': lambda is_prms: ['promisc', ['off', 'on'][is_prms]],
+        'txqueuelen': lambda qlen: ['txqueuelen', str(qlen)],
+        'state': lambda state: [state],
+    }
+
+    def __init__(self, module):
+        self.module = module
+        self.check_mode = module.check_mode
+        self.knobs = {}
+        for knob in self.knob_cmds.keys():
+            self.knobs[knob] = module.params[knob]
+        for param in self.params_list:
+            setattr(self, param, module.params[param])
+        if self.name and self.group_id:
+            self.module.fail_json(
+                msg=to_text('Can not use name and group_id together')
+            )
+        if self.knobs['group'] and self.group_id:
+            self.module.fail_json(
+                msg=to_text('Can not group and grop_id together')
+            )
+        if not (self.name or self.group_id):
+            self.module.fail_json(
+                msg=to_text('Either name or group_id should be present')
+            )
+        if self.namespace == self.netns and self.namespace:
```

Comment:
Rather than saying they're not allowed to be the same, treat this as a no-op (nothing needs to change)

Reply:
Great idea, thank you! I've changed it:
```
    def _netns_need_to_move(self):
        if self.namespace == self.netns:
            return False
```